cpp/oneapi/dal/backend/primitives/blas/test/gemv_dpc.cpp

```diff
@@ -0,0 +1,174 @@
+    }
+
+    auto Y() {
+        check_if_initialized();
+        return ndarray<float_t, 1>::empty(this->get_queue(), { m_ });
+    }
+
+    auto Y_ones() {
+        check_if_initialized();
+        return ndarray<float_t, 1>::ones(this->get_queue(), { m_ });
+    }
+
+    void test_gemv() {
+        auto y = Y();
+        auto [y_ones, y_ones_e] = Y_ones();
+        auto [a, a_e] = A();
+        auto [x, x_e] = X();
+        auto [at, at_e] = At();
+
+        SECTION("Ax") {
+            gemv(this->get_queue(), a, x, y, { a_e, x_e }).wait_and_throw();
+            check_value_vector(y, m_, n_);
+        }
+
+        SECTION("At x") {
+            gemv(this->get_queue(), at.t(), x, y, { at_e, x_e }).wait_and_throw();
+            check_value_vector(y, m_, n_);
+        }
+
+        SECTION("Ax alpha + beta") {
+            const float_t alpha = 2;
+            const float_t beta = 3;
+
+            gemv(this->get_queue(), a, x, y_ones, alpha, beta, { a_e, x_e, y_ones_e })
+                .wait_and_throw();
+            check_value_vector(y_ones, m_, n_ * alpha + beta);
+        }
+
+        SECTION("At x alpha + beta") {
+            const float_t alpha = 2;
+            const float_t beta = 3;
+
+            gemv(this->get_queue(), at.t(), x, y_ones, alpha, beta, { a_e, x_e, y_ones_e })
+                .wait_and_throw();
+            check_value_vector(y_ones, m_, n_ * alpha + beta);
+        }
+    }
+
+    void check_value_vector(const ndarray<float_t, 1>& vec, int64_t len, float_t value) {
+        check_if_initialized();
+        // const float_t* ptr2 = vec.get_data();
+        CAPTURE(vec.get_shape()[0]);
+        REQUIRE(vec.get_shape() == ndshape<1>{ len });
+        const float_t* ptr = vec.get_data();
+        for (std::int64_t i = 0; i < vec.get_count(); i++) {
+            if (std::int64_t(ptr[i]) != value) {
+                CAPTURE(i, ptr[i]);
+                FAIL();
+            }
+        }
+        SUCCEED();
+    }
+
+    bool is_initialized() const {
+        return m_ > 0 && n_ > 0;
+    }
+
+    void check_if_initialized() {
+        if (!is_initialized()) {
+            throw std::runtime_error{ "gemv test is not initialized" };
+        }
+    }
+
+private:
+    std::int64_t m_;
+    std::int64_t n_;
+};
+
+using gemv_types = COMBINE_TYPES((float, double), (c_order, f_order));
+
+TEMPLATE_LIST_TEST_M(gemv_test, "ones matrix gemv on small sizes", "[gemv][small]", gemv_types) {
+    // DPC++ GEMM from micro MKL libs is not supported on GPU
+    SKIP_IF(this->get_policy().is_cpu());
+
+    // Test takes too long time if HW emulates float64
+    SKIP_IF(this->not_float64_friendly());
+
+    this->generate_small_dimensions();
+    this->test_gemv();
+}
+
+TEMPLATE_LIST_TEST_M(gemv_test, "ones matrix gemv on medium sizes", "[gemv][small]", gemv_types) {
+    // DPC++ GEMM from micro MKL libs is not supported on GPU
+    SKIP_IF(this->get_policy().is_cpu());
```

Comment:
It is not supported on GPU but `SKIP_IF(... .is_cpu())`?

Reply:
> It is not supported on GPU but `SKIP_IF(... .is_cpu())`?

Nope. It is not. AFAIK on CPU if should fall back to the conventional MKL GEMV implementation